forms/gridfield/GridField.php

```diff
@@ -894,21 +1107,22 @@ public function __construct(GridField $gridField, $name, $title, $actionName, $a
 	}
 
 	/**
-	 * urlencode encodes less characters in percent form than we need - we
-	 * need everything that isn't a \w.
+	 * Encode all non-word characters.
+	 *
+	 * @param string $value
 	 *
-	 * @param string $val
+	 * @return string
 	 */
-	public function nameEncode($val) {
-		return preg_replace_callback('/[^\w]/', array($this, '_nameEncode'), $val);
+	public function nameEncode($value) {
+		return (string) preg_replace_callback('/[^\w]/', array($this, '_nameEncode'), $value);
 	}
 
 	/**
-	 * The callback for nameEncode
+	 * @param array $match
 	 *
-	 * @param string $val
+	 * @return string
 	 */
-	public function _nameEncode($match) {
+	public function _nameEncode(array $match) {
```

Comment:
Is this an explicit API change of a public method? Probably should remove that for now.
Why are you deleting the method description?